src/lib/api/attachment.ts

```diff
@@ -20,8 +22,13 @@ export async function getAttachmentUrl(attachment: { key: string }): Promise<str
 
 export function download(key: string): void {
   (async () => {
-    const url = await getAttachmentUrl({ key });
+    // This works, it's using the `https://res.cloudinary.url/..` to display the file
+    // (eg. this is how it looks like: https://res.cloudinary.com/fact0ry-dev/image/upload/v1687385590/ax9wzrns7msfezo0ziqu.pdf)
+    // but zero-web is uploading files to S3 first, and this api (getAttachmentUrl) was
+    // created to get the file from S3, should we do that here as well?
```

Comment:
hmm... Maybe Joel will have some context here. I think we probably should use S3 for non-image attachments? or maybe all attachments? Cloudinary is more for web served images.

Reply:
I checked a little more, and in zOS we're sending the "native file" directly to this route https://github.com/m3m3n70/zero-api/blob/7c96d9646ad1e0b779017821eccdf6a43ff3af27/src/routes/upload.ts#L85

This is uploading the file first to cloudinary, and then doing a `sendFileMessage` via sendbird. 

I think if we're going to refactor this, then i think we upload to s3 first, and then use the `signedUrl` directly & send to this route - https://github.com/m3m3n70/zero-api/blob/development/src/routes/chatChannels.ts#L66 i believe?

Reply:
i haven't looked at what we're doing in zos. but in zero, when you add a file to a message, we figure out what type of file it is. if it's an image or video, we upload to cloudinary. if it's any other type of file, it becomes an 'attachment', which gets uploaded to S3.

Reply:
this is where the type is determined:
https://github.com/m3m3n70/zero-web/blob/development/src/components/message-input/index.tsx#L189

this is where we upload non-media "attachments":
https://github.com/m3m3n70/zero-web/blob/development/src/app/api/file/index.ts#L98

Reply:
updated to upload to s3 first, and then call the api.